proposals/target-typed-new.md

```diff
@@ -38,52 +41,42 @@ object_creation_expression
 
-- **Any struct type**
-- **Any reference type**
+- **Any struct type** (including tuple types)
+- **Any reference type** (including delegate types)
 - **Any type parameter** with a constructor or a `struct` constraint
 
 with the following exceptions:
 
 - **Enum types:** not all enum types contain the constant zero, so it should be desirable to use the explicit enum member.
 - **Interface types:** this is a niche feature and it should be preferable to explicitly mention the type.
 - **Array types:** arrays need a special syntax to provide the length.
-- **Struct default constructor**: this rules out all primitive types and most value types. If you wanted to use the default value of such types you could write `default` instead.
+- **dynamic:** we don't allow `new dynamic()`, so we don't allow `new()` with `dynamic` as a target type.
 
 All the other types that are not permitted in the *object_creation_expression* are excluded as well, for instance, pointer types.
 
-> **Open Issue:** should we allow delegates and tuples as the target-type?
-
-The above rules include delegates (a reference type) and tuples (a struct type). Although both types are constructible, if the type is inferrable, an anonymous function or a tuple literal can already be used.
-```cs
-(int a, int b) t = new(1, 2); // "new" is redundant
-Action a = new(() => {}); // "new" is redundant
-
-(int a, int b) t = new(); // ruled out by "use of struct default constructor"
-Action a = new(); // no constructor found
+When the target type is a nullable value type, the target-typed `new` will be converted to the underlying type instead of the nullable type.
 
-var x = new() == (1, 2); // ruled out by "use of struct default constructor"
-var x = new(1, 2) == (1, 2) // "new" is redundant
-```
+### Miscellaneous
 
+`throw new()` is disallowed.
 
-> **Open Issue:** should we allow `throw new()` with `Exception` as the target-type?
+Target-typed `new` is allowed with binary operators, where the target type is determined by overload resolution. This excludes tuple equality.
 
-We have `throw null` today, but not `throw default` (though it would have the same effect). On the other hand, `throw new()` could be actually useful as a shorthand for `throw new Exception(...)`. Note that it is already allowed by the current specification. `Exception` is a reference type, and the specification for the throw statement says that the expression is converted to `Exception`.
+It is disallowed when there is no type to target: unary operators, collection of a `foreach`, in a `using`, in a deconstruction, in an `await` expression, as an anonymous type property (`new { Prop = new() }`), in a `lock` statement, in a `sizeof`, in a `fixed` statement, in a member access (`new().field`), in a dynamically dispatched operation (`someDynamic.Method(new())`), in a LINQ query, as the operand of the `is` operator, as the left operand of the `??` operator,  ...
```

Comment:
Should we include `goto case new()` here? to my surprise, `goto case default` works.

Reply:
`goto case` has a target type (the type of the switch expression).  In this case it doesn't work because it isn't a constant expression.

Reply:
To clarify my point, it's now forbidden to write `case default:` to avoid confusion with `default:` case. Same reasoning could be applied to `goto default;` and `goto case default;`.

> it doesn't work because it isn't a constant expression.

It could work because we now permit `new()` with primitive types, per ldm.

Anywho, I think we should permit `goto case new()` iff we permit `case new()`.